crates/sui-faucet/src/faucet/simple_faucet.rs

```diff
@@ -87,24 +88,77 @@ impl SimpleFaucet {
         })
     }
 
-    async fn select_coins(&self, number_of_coins: usize) -> Vec<ObjectID> {
+    async fn select_coins(
+        &self,
+        number_of_coins: usize,
+        transfer_amount: u64,
+    ) -> anyhow::Result<Vec<ObjectID>> {
         assert!(number_of_coins > 0);
         // If the gas candidate queue is exhausted, the request will be
```

Comment:
I might have missed it, iirc we will have 1/10 of all SUI supply here, do we want to add an error when all supply is exhausted?

Reply:
yes we should!  There are other types of things we should monitor here, e.g. when not enough coins are running in low watermark balances etc. Will leave these to future PRs though